app/src/main/java/org/wikipedia/views/SuggestedArticleDescriptionsDialog.kt

```diff
@@ -42,6 +48,12 @@ class SuggestedArticleDescriptionsDialog(
         binding.secondSuggestionFlag.setOnClickListener {
             SuggestedArticleDescriptionsReportDialog(context, binding.firstSuggestion.text.toString()) { dismiss() }.show()
         }
+
+        setOnDismissListener {
+            if (!suggestionChosen) {
+                MachineGeneratedArticleDescriptionsAnalyticsHelper.machineGeneratedSuggestionsDialogDismissed(context)
```

Comment:
Is this expected that send breadcrumb data only if no suggestion was chosen? Feels like this `dismiss` action should be sent whenever the suggestion is selected or not.

Reply:
The `Dismiss event needs to be sent only when the dialog is `cancelled` or dismissed because of touch outside the dislog window... but the callback is triggered even when the user chooses a suggestion, which it shouldn't so excluding only that condition from the callback

Reply:
I see, thanks! I think the word "DialogDismissed" looks a little bit confusing to me, not sure if there has any other better word to describe it?

Reply:
`OptedOut` it is !